substrate/frame/scheduler/src/lib.rs

```diff
@@ -1137,18 +1248,31 @@ impl<T: Config> Pallet<T> {
 					}
 					let wake = now.saturating_add(period);
 					match Self::place_task(wake, task) {
-						Ok(_) => {},
+						Ok(new_address) => {
+							if let Some(RetryConfig { total_retries, remaining, period }) =
+								Retries::<T>::take((when, agenda_index))
```

Comment:
if current iteration for periodic failed and retry scheduled, is not this always `none`?

In description you mention that next periodic is placed on hold. What about just letting a user to decide? If a user does not want them overlap, a user controls it with a number of `retries`  and `period`. A user might not want their periodic tasks to be placed on hold, because of retry, to me retry has a lower priority. 
Also after all retries attempts, it can be too late to schedule next periodic iteration, I think in this case we loose the periodic task. 
Without the hold, the logic looks more straightforward to me. As I mentioned in the issue, I think users should assess the possibility of overlapping between a periodic task and retries on their own.

Reply:
> if current iteration for periodic failed and retry scheduled, is not this always none?

If that happens, the function exits early here. But there is this [corner case](https://github.com/paritytech/polkadot-sdk/pull/3060/files#diff-88e61363c8fccbe1b7de6ee1c6dbc2afdb4e74e2aebab1f151953be8e2ec87ceR1326) where there isn't enough weight to do the retry logic, and in that case retries are ignored, test [here](https://github.com/paritytech/polkadot-sdk/pull/3060/files#diff-0d54d132118a2b638382f5259175174c3bbcc6919259fc5bc4262e7e1fff67efR1076).

Reply:
Regarding the rest of your comment, I can see why you'd want the period and retry to work independently. I'm ok to make it do this:
- if a task fails, try to schedule a retry if a retry config is set
- if the task is periodic, try to schedule the next iteration

I think if we make this behavior configurable (i.e. let users decide which behaviour they want), it would get way too complicated and it's not worth it. I think it's better if it's simpler.

> I think users should assess the possibility of overlapping between a periodic task and retries on their own

I was trying to avoid this because I saw it as a footgun, but I think you're right. If users set the configurations correctly, it shouldn't be an issue.

Reply:
Proposed changes in https://github.com/paritytech/polkadot-sdk/pull/3060/commits/9da58c6eb6eda067ce673f46485d2c3738293d7c